### PR TITLE
fixed ban new button issue 

### PR DESCRIPTION
**Pull request by darshanvss**
Banned competitors page bugs #9652

- Moved 'No Data to show' logic into BannedCompetitors.jsx
- Moved the 'Ban new competitor' button to top of list
- Added new boolean prop to have only one 'Ban new Competitor' button

### CODE REVIEW

**Comment by darshanvss** [comment]
The formatting is being changed when I am saving files as I am using prettier. How do I avoid this so it would be easier to see the actual changes?

File: app/webpacker/components/Panel/pages/BannedCompetitorsPage/BannedCompetitors.jsx [hunk 1]
```diff
@@ -1,67 +1,78 @@
...
-              </Table.Cell>
-              <Table.Cell>{role.start_date}</Table.Cell>
-              <Table.Cell>{role.end_date}</Table.Cell>
-              <Table.Cell>{role.metadata.ban_reason}</Table.Cell>
-              <Table.Cell>{role.metadata.scope}</Table.Cell>
+      {canEditBannedCompetitors && !isPastBannedCompetitorList && (
+        <Button onClick={() => setBanModalParams({ action: "new" })}>
+          Ban new competitor
+        </Button>
+      )}
```
Comment (danieljames-dj) [inline app/webpacker/components/Panel/pages/BannedCompetitorsPage/BannedCompetitors.jsx:20]: This button can be moved to the parent component (BanendCompetitorsPage/index.jsx). This way, you don't have to pass the `isPastBannedCompetitorList` boolean.

The main challenge of moving this to parent component will be that you might have to move the modal and related states as well, and then pass the `editBannedCompetitor` action as a parameter to this component, and you can call that parameter as action for the edit.

Hope this is clear, let me know if you need more clarifications.

**Review by danieljames-dj — COMMENTED** [review]
Thanks a lot @darshanvss for taking the initiative to work on this.

I've left some comments. Also, regarding prettier, I would suggest to disable it for this project. The reason is that different people can have different prettier configurations and we don't have a prettierrc in our code to make it unique.

I personally use eslint for formatting because eslint will use the configs in `eslintrc.json` of our repository and hence there won't be any conflicts from existing codebase.

File: app/webpacker/components/Panel/pages/BannedCompetitorsPage/BannedCompetitors.jsx (hunk 1 above)
Comment (darshanvss) [reply app/webpacker/components/Panel/pages/BannedCompetitorsPage/BannedCompetitors.jsx]: @danieljames-dj I have moved the ban new competitor button and it's modal to the parent comp.  For linting,  I am running `npx eslint /path_to_file/file.jsx` but is asking to install eslint. Is that fine to insall/update packages and push code? I have not run it

Comment (danieljames-dj) [reply app/webpacker/components/Panel/pages/BannedCompetitorsPage/BannedCompetitors.jsx]: Thanks a lot @darshanvss. It's fine to install eslint I guess. It's up to you. Anyway, the current changes looks great to me.

**Review by danieljames-dj — APPROVED** [review]

File: app/webpacker/components/Panel/pages/BannedCompetitorsPage/BannedCompetitors.jsx (hunk 1 above)
Comment (gregorbg) [reply app/webpacker/components/Panel/pages/BannedCompetitorsPage/BannedCompetitors.jsx]: Re eslint: The issue is the package manager. For some reason, your environment wants to run `npx` which is a script executor for NodeJS. However, we are using **Yarn** to install and link our packages in JS, so you should run eslint through Yarn as well.

**Comment by danieljames-dj** [comment]
@darshanvss Can you also please fix the "TrailingWhitespace" error that is thrown?

**Comment by darshanvss** [comment]
Hi @danieljames-dj , can you tell where is the error being thrown/ how to replicate it? I am not able to get this error

**Comment by darshanvss** [comment]
Hi @danieljames-dj I have removed the whitespace 👍

**Pull request merged**